fendermint/vm/interpreter/src/fvm/gas/eip1559.rs

```diff
@@ -0,0 +1,108 @@
+use crate::fvm::gas::{Gas, GasMarket};
+use crate::fvm::state::{read_actor_state, FvmExecState};
+use crate::fvm::FvmMessage;
+use anyhow::anyhow;
+use fendermint_actor_gas_market::EIP1559GasState;
+use fendermint_vm_actor_interface::gas::GAS_ACTOR_ADDR;
+use fendermint_vm_actor_interface::system;
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+type AtomicGas = AtomicU64;
+
+/// The gas market based on EIP1155
+#[derive(Default)]
+pub struct EIP1559GasMarket {
```

Comment:
The actual gas strategy (EIP1559) should be fully encapsulated in the actor, so this is just an `ActorGasMarket` (a gas market driven by an on-chain actor), i.e. a gas market that resorts to an on-chain actor to keep its state and oscillate the gas parameters.